### ISSUE TITLE
Add TripleA to Steam

### ISSUE THREAD

**Issue by ron-murhammer** [triplea-game/triplea#3434]
So this has been discussed a few times over the years but now that more open source games have actually gotten on Steam I think its probably more realistic. Battle for Wesnoth which along with TripleA were 2 of the most popular TBS games on Sourceforge back in the day has just this past month gotten added to Steam.

I'd first like to discuss whether folks think they like to see it added to Steam and second come up with a list of things we know would need to be done in order for it to be added to Steam.

**Comment by RoiEXLab** [comment]
I'm not familiar with the build process involved, but I see a couple of problems:
1. Stable releases: Our current way of stable releasing any depends on the chemark for our releases which doesn't trigger any event whenever this is changed. So we'd need to setup some sort of cronjob (perhaps using travis) which regularly checks if there's an update available and pushes it to steam.
2. Installing mechanism: We probably don't want to prompt users installing over steam with an installer. So we'd either need to invoke the current installer using some command line flags to run in the background and place the game files in the according directory. 
3. Multiple Processes: Currently we spin up a new JVM whenever we host a game or join one. Not sure if steam likes them or just completely ignores them. 
4. There's no greenlight anymore, so we'd need to have to pay the one time 100$ fee.
5. Licensing: GPL is incompatible with steam https://partner.steamgames.com/doc/sdk/uploading/distributing_opensource

TL;DR: Unlikely we're going to solve the latter issue.

**Comment by ron-murhammer** [comment]
@RoiEXLab Yeah, so point 5 was the main obstacle I always thought but I read through a bunch of articles and forums for other GPL games that have gotten on steam and it appears that as long as we don't use features provided by Steam's API which is proprietary then it is allowed: https://steamcommunity.com/workshop/filedetails/discussion/729138129/353915309336126754/
https://forums.wesnoth.org/viewtopic.php?f=6&t=44366

Not worried about point 4 as if we are fairly confident we can eventually get it distributed on Steam then it would easily be worth it. The rest of the points are things that we'd have to research but could make adjustments to if necessary. 1 & 2 should be fairly easy to make adjustments to. No idea about 3 at the moment.

A few other things I believe we'd have to do or at least consider:
6. Trailer and screenshots
7. Finishing game setup menus overhaul
8. Form some sort of organization to represent TripleA to Valve

**Comment by RoiEXLab** [comment]
@ron-murhammer 100% agree.
For Point 3 have a look at this PR: https://github.com/triplea-game/triplea/pull/2665
It basically tells us what needs to be done in order to avoid having to spin up new proccesses.

**Comment by DanVanAtta** [comment]
> 5. Licensing: GPL is incompatible with steam https://partner.steamgames.com/doc/sdk/uploading/distributing_opensource

This is a deal breaker, no?

**Comment by ron-murhammer** [comment]
@DanVanAtta No, read my post right after that.

**Comment by Cernelius** [comment]
And this?
https://www.moddb.com/games/triplea
I don't know if there is a list somewhere of all external places where TripleA is added to.

**Comment by DanVanAtta** [comment]
@Cernelius been some time since moddb was last updated. That might be one where everyone had lost the credentials : |

We used to have a list of a half dozen mirrors/sites to update; all-in-all one reason or another we can't update them or they were not worth updating (close to zero traffic from the sites). At this point the only website consistently hosting a link to the latest is our own. 

There are some which we do not control out there, but from what I can tell they have migrated, at least the ones that are active.

**Comment by Cernelius** [comment]
Not up to me, but I would rather remove what is not going to be updated, if possible.

If exhaustive, what about adding that list to the Wiki here.

**Comment by DanVanAtta** [comment]
@Cernelius The wiki link does not seem to have come through.

I would agree we should do something about out of date sites, not necessarily just modDb. Would you help in an effort to try and identify out of date mirrors?  

Let's not crowd out this issue and follow up in: https://github.com/triplea-game/triplea/issues/3461

**Comment by General-Dru-Zod** [comment]
I like the idea of more exposure and downloadable from other sites. Or at the very least have links on other sites directing them to us.

**Comment by ron-murhammer** [comment]
@DanVanAtta I'd like to get through the menu overhaul that @RoiEXLab started so TripleA looks a bit more polished and less like a 90s swing app. Probably just leave this for now.

**Comment by RoiEXLab** [comment]
@ron-murhammer I'll try working on it again in October

**Comment by RoiEXLab** [comment]
Implementing UI is harder than initially anticipated.
There are just too many core components dependent on swing that really shouldn't be.
I'll try to focus on those in my next PRs.
I will try to get rid of Observable (deprecated as of java 9) and overhaul the ISetupPanel interface to not require a JComponent to be returned.

**Comment by RoiEXLab** [comment]
I'd suggest to ice box this issue until we have a better UI in place

**Comment by ssoloff** [comment]
> I will try to get rid of Observable (deprecated as of java 9)

@RoiEXLab Speaking of which ...  Do you know what happened to those deprecation warnings in the Java 9 build?  They just seem to have disappeared recently.

**Comment by RoiEXLab** [comment]
@ssoloff No, didn't even notice they were gone. They should be somewhere though

**Comment by DanVanAtta** [comment]
Let's re-open when we're ready to pick this topic back up. I think we're all in pretty strong support of TripleA on Steam, there is a lot to do before then :sweat_smile:
